var/spack/repos/builtin/packages/gmsh/package.py

```diff
@@ -66,8 +66,7 @@ class Gmsh(Package):
     depends_on('fltk', when='+fltk')
     depends_on('hdf5', when='+hdf5')
     depends_on('oce',  when='+oce')
-    depends_on('petsc+mpi', when='+petsc+mpi')
-    depends_on('petsc', when='+petsc~mpi')
+    depends_on('petsc', when='+petsc')
```

Comment:
here you allow to build `gmsh+mpi` against `petsc~mpi`, I don't think it's a correct behaviour.